alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/util/DestinationType.kt

```diff
@@ -18,11 +18,16 @@ package com.amazon.opendistroforelasticsearch.alerting.util
 enum class DestinationType(val value: String) {
     CHIME("chime"),
     SLACK("slack"),
+    SNS("sns"),
     CUSTOM_WEBHOOK("custom_webhook"),
     EMAIL("email"),
     TEST_ACTION("test_action");
 
     override fun toString(): String {
         return value
     }
+
+    companion object {
+        var snsUseIamRole = false
```

Comment:
This seems a little strange to me. I'm not sure if we should maintain state on using IAM role in the `DestinationType` enum class.

It looks like this is mostly used for null assertion in the `SNS` data class. Perhaps you can remove this variable and instead validate if either `roleARN` or `access/secret` is set somewhere. It looks like you something like that in `SNSMessage` already, that might be sufficient.

And ideally, we should allow the `access/secret` settings to be changed using the reload API and if we support that then you can have scenarios where:

1. `access/secret` settings are originally set
2. SNS Destinations are created (allowing `roleARN` to be null since `snsUseIamRole = false`)
3. The `access/secret` settings are removed, checking `roleARN` on any subsequent SNS Destination but now you have one that's already created with a null `roleARN` and no `access/secret` settings so it bypassed this check.

Reply:
I need this in case the customer uses the indexDestination API to create the SNS destination and we need some validation for the role arn.
Also I have updated the code such that `snsUseIamRole` can change whenever the settings are reloaded.